### PR TITLE
[train_unconditional] fix gradient accumulation.

### PR DESCRIPTION
**Pull request by patil-suraj**
The training script uses `accelerator.accumulate` for gradient accumulation but the `gradient_accumulation_steps` arg is not passed when creating the `Accelerator`. Also the `global_step` should be updated only when the gradients are synchronised.

### CODE REVIEW

**Review by anton-l — APPROVED** [review]
Wow, the missing `gradient_accumulation_steps` must've been lost in a merge resolution, thanks a lot for spotting that, and for the step update solution!
